### DISCUSSION TITLE
airflow upgrading failed apache-airflow==2.2.5 to apache-airflow[celery]==2.7.0

### DISCUSSION THREAD

**Discussion by Sathya-omnius** [apache/airflow#33728] [Q&A]
HI Team, 
when I am trying to upgrade Airflow I am getting following errors
export AIRFLOW_CONN_AIRFLOW_DB="postgresql+psycopg2://${DATABASE_USERNAME}:${DATABASE_PASSWORD}@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_NAME}?options=-csearch_path=${DATABASE_NAME}"
echo "postgresql+psycopg2://${DATABASE_USERNAME}:***@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_NAME}"
echo RUNNING airflow db init
airflow db init
echo RUNNING airflow db migrate
airflow db migrate
echo RUNNING airflow connections delete airflow_db
airflow connections delete airflow_db
echo RUNNING airflow connections add airflow_db
airflow connections add airflow_db --conn-uri "$AIRFLOW_CONN_AIRFLOW_DB"

Erros I am getting on the pod:
cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "INCLUDE"
LINE 1: ...rt_date ON task_instance (dag_id, task_id, state) INCLUDE (s...
                                                             ^
[SQL: CREATE INDEX ti_state_incl_start_date ON task_instance (dag_id, task_id, state) INCLUDE (start_date)]
(Background on this error at: https://sqlalche.me/e/14/f405)
RUNNING airflow connections delete airflow_db
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:78: DeprecationWarning: The log_format option in [core] has been moved to the log_format option in [logging] - the old setting has been used, but please update your config.
  LOG_FORMAT = conf.get("logging", "log_format")
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:79: DeprecationWarning: The simple_log_format option in [core] has been moved to the simple_log_format option in [logging] - the old setting has been used, but please update your config.
  SIMPLE_LOG_FORMAT = conf.get("logging", "simple_log_format")
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:587 DeprecationWarning: The hide_sensitive_variable_fields option in [admin] has been moved to the hide_sensitive_var_conn_fields option in [core] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:830 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:817 DeprecationWarning: The logging_level option in [core] has been moved to the logging_level option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:818 DeprecationWarning: The logging_level option in [core] has been moved to the logging_level option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:817 DeprecationWarning: The fab_logging_level option in [core] has been moved to the fab_logging_level option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:818 DeprecationWarning: The fab_logging_level option in [core] has been moved to the fab_logging_level option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:751 UserWarning: Config scheduler.max_tis_per_query (value: 512) should NOT be greater than core.parallelism (value: 120). Will now use core.parallelism as the max task instances per query instead of specified value.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:857 FutureWarning: The 'navbar_color' setting in [webserver] has the old default value of '#007A87'. This value has been changed to '#fff' in the running config, but please update your config before Apache Airflow 2.1.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:857 FutureWarning: The 'dag_default_view' setting in [webserver] has the old default value of 'tree'. This value has been changed to 'grid' in the running config, but please update your config before Apache Airflow 3.0.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:723 DeprecationWarning: The log_filename_template option in [core] has been moved to the log_filename_template option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:857 FutureWarning: The 'log_filename_template' setting in [logging] has the old default value of '{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log'. This value has been changed to 'dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log' in the running config, but please update your config before Apache Airflow 3.0.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:723 DeprecationWarning: The auth_backend option in [api] has been renamed to auth_backends - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:765 DeprecationWarning: The auth_backend option in [api] has been renamed to auth_backends - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:737 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/logging_config.py:35 DeprecationWarning: The logging_config_class option in [core] has been moved to the logging_config_class option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/config_templates/airflow_local_settings.py:29 DeprecationWarning: The logging_level option in [core] has been moved to the logging_level option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/config_templates/airflow_local_settings.py:34 DeprecationWarning: The fab_logging_level option in [core] has been moved to the fab_logging_level option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/config_templates/airflow_local_settings.py:36 DeprecationWarning: The log_format option in [core] has been moved to the log_format option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/config_templates/airflow_local_settings.py:51 DeprecationWarning: The base_log_folder option in [core] has been moved to the base_log_folder option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/config_templates/airflow_local_settings.py:64 DeprecationWarning: The log_processor_filename_template option in [core] has been moved to the log_processor_filename_template option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/config_templates/airflow_local_settings.py:193 DeprecationWarning: The remote_logging option in [core] has been moved to the remote_logging option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/logging_config.py:84 DeprecationWarning: The task_log_reader option in [core] has been moved to the task_log_reader option in [logging] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:283 DeprecationWarning: The sql_alchemy_pool_enabled option in [core] has been moved to the sql_alchemy_pool_enabled option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:290 DeprecationWarning: The sql_alchemy_pool_size option in [core] has been moved to the sql_alchemy_pool_size option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:308 DeprecationWarning: The sql_alchemy_pool_recycle option in [core] has been moved to the sql_alchemy_pool_recycle option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/settings.py:346 DeprecationWarning: The sql_engine_encoding option in [core] has been moved to the sql_engine_encoding option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py:974 DeprecationWarning: The namespace option in [kubernetes] has been moved to the namespace option in [kubernetes_executor] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/stats.py:51 DeprecationWarning: The statsd_on option in [scheduler] has been moved to the statsd_on option in [metrics] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/celery/executors/celery_executor.py:149 DeprecationWarning: The default_queue option in [celery] has been moved to the default_queue option in [operators] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/models/base.py:27 DeprecationWarning: The sql_alchemy_schema option in [core] has been moved to the sql_alchemy_schema option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/models/base.py:71 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/sqlalchemy.py:48 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
ERROR: You need to initialize the database. Please run `airflow db init`. Make sure the command is run using Airflow version 2.7.0.

can some one help how to fix this issue.

**Comment by potiuk** [comment]
You are likely using unsupported database or database that is not compliant with the standard it claims to be compliant. But you have not specified what database you have and from which provider (cloud?) you have.

**Reply by Sathya-omnius** [reply]
HI Potiuk, we are using postgresql with Azure cloud, AIrflow with 2.5.3 is also working for us , but when we try to use 2.7 we are facing the errros

**Reply by potiuk** [reply]
Which postgres version on Azure ?

**Reply by potiuk** [reply]
Quite likely you are using Postgresql 10 which does not support INCLUDE INDEX. Airflow 2.7.0 supports Postgres 11+. Postgres 10 is already not supported by Postgres as it reached end of life many months ago. 

If you used Postgres 10 for Airflow 2.5, you were lucky because generally we dropped support and stopped testing it with Postgres 10 in 2.5.0 - so it worked "accidentally" for you.

You should upgrade to Postgres 11 at least to use Airflow 2.7.

Here are prerequisites.

Here are prerequisites: https://airflow.apache.org/docs/apache-airflow/stable/installation/prerequisites.html

**Reply by Sathya-omnius** [reply]
HI Potiuk,

thanks for your valuable inputs
